server/services/v1/realtime.go

```diff
@@ -103,35 +110,81 @@ func (s *realtimeService) DeviceConnectionHandler(w http.ResponseWriter, r *http
 	if err != nil {
 		// ToDo: Change to WS errors
 		w.WriteHeader(http.StatusInternalServerError)
-		_, _ = w.Write([]byte(fmt.Sprintf(`{"error": "%s"}`, err.Error())))
+		_, _ = w.Write([]byte(fmt.Sprintf(`{"event_type": 2, "event": {"code": 1011, "message": "%s"}}`, err.Error())))
 		return
 	}
 
-	log.Debug().Msgf("params '%v'", params)
-	_ = conn.WriteMessage(websocket.TextMessage, []byte(`{"error": "not implemented"}`))
-	_ = conn.Close()
+	ctx := r.Context()
+	session, err := s.devices.AddDevice(ctx, conn, params)
+	if err != nil {
+		realtime.SendReply(conn, params.ToEncodingType(), api.EventType_error, errors.InternalWS(err.Error()))
```

Comment:
This needs to be a RealtimeMessage type otherwise the client cannot decode it.

Reply:
this is updated in another diff, will rebase now.